exercises-toolbox/2-numpy/2-indexing/loesung.py

```diff
@@ -22,8 +22,10 @@
 print("Spielfeld zu Beginn des Spiels:")
 print(field)
 
-mat = plt.matshow(field, cmap="Set1", vmax=5, vmin=-1)
-plt.savefig("Spielfeld_Beginn.pdf")
+fig, ax = plt.subplots(1, 1, constrained_layout=True)
+mat = ax.matshow(field, cmap="Set1", vmax=5, vmin=-1)
+fig.savefig("Spielfeld_Beginn.pdf")
+ax.cla()
```

Comment:
Why cla?

Reply:
Because I want to clear the axes and keep the figure to plot the next matshow instead of clearing the whole figure and opening a new one.
clf vs cla

Of course we could also use `fig.clf()` and then do a `ax = fig.add_subplot()` and continue with a new set of axes. That works as well but is a bit more typing to do and I went for the other way.
Thoughts on why to not use that approach?

Reply:
As it is in the solution for a numpy exercise, I think we can keep it that way. Those who read this line will probably be the ones leaning towards a deeper understanding and will see a different way to handle multiple figures.

Reply:
> Because I want to clear the axes and keep the figure to plot the next matshow instead of clearing the whole figure and opening a new one.
clf vs cla

But you only update the matrix, the rest stays the same, so erasing the whole axes is overkill.

Instead of calling `ax.cla()`, just update the matrix of the `matshow` plot:

```
mat.set_array(field)
fig.savefig("Spielfeld_Endle.pdf")
```